cpu/arm7_common/Kconfig

```diff
@@ -11,7 +11,7 @@ config CPU_ARCH_ARMV4T
     select HAS_ARCH_ARM
     select HAS_CPP
     select HAS_LIBSTDCPP
-    select MODULE_MALLOC_THREAD_SAFE if TEST_KCONFIG
+    select MODULE_MALLOC_THREAD_SAFE
```

Comment:
Yeah, the problem was that you removed the `if TEST_KCONFIG` on all locations, but this symbol, for instance, is not a module and does not depend on `TEST_KCONFIG`. In my previous comment I meant just that only occurrence, because the selecting symbol (a module) already depended on `TEST_KCONFIG`.

Reply:
thanks! Lets see if it passes this time